state/userAccount.go

```diff
@@ -218,12 +220,16 @@ func (a *userAccount) GetAllLeaves(
 	leavesChannels *common.TrieIteratorChannels,
 	ctx context.Context,
 ) error {
-	dataTrie := a.dataTrieTracker.DataTrie()
-	if check.IfNil(dataTrie) {
+	if check.IfNil(a.dataTrieInteractor) {
+		return ErrNilTrackableDataTrie
```

Comment:
is this dead code? Can we have a situation when we end up with a non-nil userAccount but nil dataTrieInteractor?

Reply:
If some bytes are directly unmarshalled in a `&userAccount`, that user account will have a nil `trackableDataTrie`. Will move the `userAccount` in a separate package. This way, a `userAccount` will be created only by using the constructor. Then, these checks can be safely removed.

Reply:
Added TODO, will solve in the next PR.